feat: add module "On this page"

sanua356 opened the pull request:
Added a component that allows you to view and navigate between page headings. Mobile and desktop versions.

Desktop: 
![image](https://github.com/user-attachments/assets/b32c9ec2-80fd-4017-af9e-af0d773f20bb)

Desktop (fullwidth page editor):
![image](https://github.com/user-attachments/assets/12322874-606c-40ba-a1ae-db63e7897f3e)


Mobile:
![image](https://github.com/user-attachments/assets/7111da92-f8b8-44e8-b15f-2b20492688cb)

Light theme:
![image](https://github.com/user-attachments/assets/0efabeda-b82b-435b-b488-9470aa20d89a)

sanua356 committed:
feat: add table of contents in frontend

sanua356 committed:
chore: add scroll to heading

sanua356 committed:
chore: add mobile menu

sanua356 committed:
chore: add backend support

sanua356 committed:
remove console log

sanua356 committed:
fix conflicts

sanua356 committed:
Merge branch 'main' into table-of-contents

sanua356 committed:
Merge branch 'main' into table-of-contents

sanua356 committed:
fix: change lib

sanua356 committed:
Merge branch 'main' into table-of-contents

sanua356 committed:
Merge branch 'main' into table-of-contents

sanua356 committed:
fix: delete console log

sanua356 committed:
fix: fix nullish preferences

sanua356 committed:
fix: nullish preferences api error

sanua356 committed:
Merge branch 'docmost:main' into table-of-contents

sanua356 committed:
rewrite code to intersection observer API

sanua356 committed:
Merge branch 'main' into table-of-contents

sanua356 committed:
chore: add fullwidth drawer

sanua356 committed:
fix: hide fullwidth headings button if settings preference disabled

sanua356 commented:
> Does this get exported?

No

sanua356 commented:
> It would be good to be able to rename the "on this page" wording 

I think it would be convenient, but at the moment I don't see the need

sanua356 committed:
Merge branch 'main' into table-of-contents

Philipinho committed:
upgrade to NestJS 11 (#766)

* upgrade to nest 11

* update dependencies

h4rb4rd committed:
fix: accept invitation password hashing (#773)

Philipinho committed:
pin s3 package to 3.701.0

Philipinho committed:
v0.8.2

Philipinho committed:
pin react-email version (#779)

Philipinho committed:
v0.8.3

Philipinho committed:
feat: standalone collab server (#767)

* feat: standalone collab server

* * custom collab server port env
* fix collab start script command

* * API prefix
* Log startup PORT

* Tweak collab debounce

Philipinho committed:
Log APP_URL on startup

Philipinho committed:
feat: add stats to standalone collab server (#798)

* Log APP_URL on startup

* add stats endpoint to standalone collab server

Philipinho committed:
enable trustProxy

Philipinho committed:
feat: sentry (#802)

Philipinho committed:
* Add SENTRY_DNS env variable
* Commit lock file

NomadRazor committed:
feat: add copy invite link to invitation action menu (#360)

* +copy invite link to clipboard from invite action menu

* -remove log to console for copy link action

* Refactor copy invite link feature

---------

Co-authored-by: Philipinho <16838612+Philipinho@users.noreply.github.com>

Philipinho committed:
push files left (#360) (#804)

Philipinho committed:
cleanup

Philipinho committed:
v0.8.4

Philipinho committed:
fix sentry

Philipinho committed:
revert sentry (#808)

* revert sentry
* remove sentry env

l2dy committed:
fix: add missing awaits (#814)

Philipinho committed:
feat: cloud and ee (#805)

* stripe init
git submodules for enterprise modules

* * Cloud billing UI - WIP
* Proxy websockets in dev mode
* Separate workspace login and creation for cloud
* Other fixes

* feat: billing (cloud)

* * add domain service
* prepare links from workspace hostname

* WIP

* Add exchange token generation
* Validate JWT token type during verification

* domain service

* add SkipTransform decorator

* * updates (server)
* add new packages
* new sso migration file

* WIP

* Fix hostname generation

* WIP

* WIP

* Reduce input error font-size
* set max password length

* jwt package

* license page - WIP

* * License management UI
* Move license key store to db

* add reflector

* SSO enforcement

* * Add default plan
* Add usePlan hook

* * Fix auth container margin in mobile
* Redirect login and home to select page in cloud

* update .gitignore

* Default to yearly

* * Trial messaging
* Handle ended trials

* Don't set to readonly on collab disconnect (Cloud)

* Refine trial (UI)
* Fix bug caused by using jotai optics atom in AppHeader component

* configurable database maximum pool

* Close SSO form on save

* wip

* sync

* Only show sign-in in cloud

* exclude base api part from workspaceId check

* close db connection beforeApplicationShutdown

* Add health/live endpoint

* clear cookie on hostname change

* reset currentUser atom

* Change text

* return 401 if workspace does not match

* feat: show user workspace list in cloud login page

* sync

* Add home path

* Prefetch to speed up queries

* * Add robots.txt
* Disallow login and forgot password routes

* wildcard user-agent

* Fix space query cache

* fix

* fix

* use space uuid for recent pages

* prefetch billing plans

* enhance license page

* sync

Philipinho committed:
Hide version in cloud

Philipinho committed:
Add api prefix to attachment nodes

Philipinho committed:
* Update EE license fil
* State license in Readme file

Philipinho committed:
* Upgrade Dockerfile to node 22
* Pin pnpm to pnpm@10.4.0

Philipinho committed:
fix

Philipinho committed:
add pnpm to packageManager for consistency

Philipinho committed:
* configurable trial days
* hide create sso provider in cloud

Philipinho committed:
sync

Philipinho committed:
fix cookie name

Philipinho committed:
remove cloud env check

Philipinho committed:
exclude /health/live endpoint

Philipinho committed:
fix space translations (#826)

Philipinho committed:
Option to log db queries in dev mode (#827)

Philipinho committed:
sort workspace list

fuscodev committed:
feat: editor emoji picker (#775)

* feat: emoji picker

* fix: lazy load emoji data

* loading animation (for slow connection)

* parsing :shortcode: and replace with emoji + add extension to title-editor

* fix

* Remove title editor support
* Remove shortcuts support
* Cleanup

---------

Co-authored-by: Philipinho <16838612+Philipinho@users.noreply.github.com>

Philipinho committed:
Add emoji command to title editor

iagocavalcante committed:
feat: adding family 6 in uri to configure for both 4 and 6 (#807)

* feat: adding family 6 in uri to configure for both 4 and 6
* feat: adding redis family in websocket config

fuscodev committed:
fix: table header in exported markdown (#769)

sanua356 committed:
chore: add print styles

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/docmost/docmost?pullRequest=625) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/docmost/docmost?pullRequest=625) before we can accept your contribution.<br/>**7** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: h4rb4rd<br/>:white_check_mark: sanua356<br/>:white_check_mark: l2dy<br/>:white_check_mark: fuscodev<br/>:white_check_mark: hoiekim<br/>:white_check_mark: GlitchDevX<br/>:white_check_mark: iagocavalcante<br/>:x: NomadRazor<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/docmost/docmost?pullRequest=625) it.</sub>

sanua356 commented:
Hello. alexander.pankratov is also my user, but he is not registered on GitHub, so I can't sign the CLA for him.

Philipinho committed:
feat: disconnect collab websocket on idle tabs (#848)

* disconnect real-time collab if user is idle
* log yjs document disconnect and unload in dev mode
* no longer set editor to read-only mode on collab websocket disconnection
* treat delayed collab websocket "connecting" state as disconnected
* increase maxDebounce to 45 seconds
* add reset handle to useIdle hook

Philipinho committed:
* prefetch sso providers in settings
* hide sso enforcement in standard plan

Philipinho committed:
exclude billing webhook endpoint

Philipinho committed:
prevent slider flickers

Philipinho committed:
Make codeblock comment more legible in light mode

Philipinho committed:
fix: VSCode markdown pasting (#857)

* fix vscode markdown pasting

* fix markdown -> html formatting

TopClans commented:
Would love to see this feature in Docmost, thanks for PR mr @sanua356! Hope it will be merged soon

Philipinho committed:
fix drag handle visibility (#868)

Philipinho committed:
feat: add page stats to page menu (#876)

Philipinho committed:
New Crowdin updates (#840)

Philipinho committed:
Add Dutch translation (#877)

Philipinho committed:
fix trial days

Philipinho committed:
fix: page title editor bugs (#892)

* Fix page title

* compare empty page title

* Properly handle null tree node name and icon

Philipinho committed:
fix

wienans commented:
@sanua356 nice work would like to see the feature.
Maybe squash and force push the branch so doublespeak the commit of the not GitHub account.

GlitchDevX committed:
feat: auto focus emoji-picker search when opened (#894)

Co-authored-by: JonasRingeis <jonas.ringeis@otto.de>

Philipinho committed:
feat: enhance editor uploads (#895)

* * multi-file paste support
* allow media files (image/videos) to be attachments
* insert trailing node if file placeholder is at the end of the editor

* fix video align

Philipinho committed:
feat: billing sync (cloud) (#899)

* Set page history to 5 minutes interval

* * Configure default queue options

* sync

* * stripe seats sync (cloud)

Philipinho committed:
Revert "feat: auto focus emoji-picker search when opened (#894)" (#900)

This reverts commit 573457403ec4c72f403917ee6f631a0a633e9e54.

Philipinho committed:
null check

Philipinho committed:
Move suspense above popover dropdown

Philipinho committed:
Refetch space list on mount

Philipinho committed:
* fix color check
* update lock file

Philipinho committed:
fix collab token refresh which leads to collab editor reconnection loop (#933)

Philipinho committed:
feat: add version check (#922)

* Add version endpoint

* version indicator

* refetch

* * Translate strings
* Handle error

hoiekim committed:
adds missing command for down migration (#908)

Philipinho committed:
telemetry module (#934)

* update lockfile

* fix color check

* telemetry

* complete

* Use interval

Philipinho committed:
update env file

Philipinho committed:
don't replace line breaks

Philipinho committed:
v0.9.0

sanua356 committed:
Merge branch 'main' into table-of-contents

sanua356 commented:
@Philipinho Hi. I rebased a branch and changed the author name of my local account in one commit that was not registered on github. Can you take a look at the branch? It's been sitting there for months.

sanua356 commented:
@Philipinho The rebase was unsuccessful. I will reopen the PR again.
